src/EbmlFloat.cpp

```diff
@@ -113,8 +113,9 @@ uint64 EbmlFloat::UpdateSize(bool bWithDefault, bool  /* bForceRender */)
 filepos_t EbmlFloat::ReadData(IOCallback & input, ScopeMode ReadFully)
 {
   if (ReadFully != SCOPE_NO_DATA) {
-    binary Buffer[20];
-    assert(GetSize() <= 20);
+    assert(GetSize() == 4 || GetSize() == 8);
+    if (GetSize() == 4 || GetSize() == 8) {
```

Comment:
Same argument here; in case of unexpected sizes the function should skip the element.